src/Raven.Client/Documents/Changes/DatabaseChanges.cs

```diff
@@ -586,7 +600,12 @@ private async Task DoWork(string nodeTag)
                         wasConnected = false;
                         try
                         {
-                            _serverNode = await _requestExecutor.HandleServerNotResponsive(_url.AbsoluteUri, _serverNode, _nodeIndex, e).ConfigureAwait(false);
+                            // If node tag is provided we should not failover to a different node
+                            // Failing over will create a mismatch if the operation is created and monitored on the provided node tag
+                            if (string.IsNullOrEmpty(nodeTag))
+                                _serverNode = await _requestExecutor.HandleServerNotResponsive(_url.AbsoluteUri, _serverNode, _nodeIndex, e).ConfigureAwait(false);
+                            else
+                                await _requestExecutor.UpdateTopologyAsync(new RequestExecutor.UpdateTopologyParameters(_serverNode) { TimeoutInMs = 0, ForceUpdate = true, DebugTag = "handle-server-not-responsive" }).ConfigureAwait(false);
```

Comment:
let's have a different `DebugTag`, so if needed we will know from which code path this request came

Reply:
done